### ISSUE TITLE
Can I manually delete partition directories under /var/lib/clickhouse/data/default/MyTable/

### ISSUE THREAD

**Issue by q0987** [yandex/ClickHouse#3531]
Hello all,

I have experienced issues while dropping tables and reported the issue here:
https://github.com/yandex/ClickHouse/issues/3497

Question> Can I manually delete OLD directory under /var/lib/clickhouse/data/default/MyTable/ while clickhouse-server is still running without causing any issues?

$ ls MyTable/ -l
total 185
drwxr-x--- 2 clickhouse clickhouse 46 Oct 26 20:08 20181026_2512_3977_458
drwxr-x--- 2 clickhouse clickhouse 46 Oct 26 20:48 20181026_3978_3985_2
drwxr-x--- 2 clickhouse clickhouse 46 Oct 29 18:45 20181029_3986_5305_617
drwxr-x--- 2 clickhouse clickhouse 46 Oct 29 18:46 20181029_5306_5310_1
drwxr-x--- 2 clickhouse clickhouse 46 Oct 31 11:30 20181030_5311_6836_1049
drwxr-x--- 2 clickhouse clickhouse 46 Oct 31 11:31 20181030_6837_6841_1
drwxr-x--- 2 clickhouse clickhouse 46 Oct 31 16:30 20181031_6842_7956_265
drwxr-x--- 2 clickhouse clickhouse 46 Oct 31 17:45 20181031_7957_7976_2
drwxr-x--- 2 clickhouse clickhouse 46 Nov  1 16:34 20181101_7977_9074_568
drwxr-x--- 2 clickhouse clickhouse 46 Nov  1 18:29 20181101_9075_9099_3
drwxr-x--- 2 clickhouse clickhouse  2 Aug 16 14:18 detached
-rw-r----- 1 clickhouse clickhouse  1 Aug 16 14:18 format_version.txt

Basically, I need to remove directories generated before 20181101_xxx_xxx_xxx.

Thank you

**Comment by 4ertus2** [comment]
Hello,

If your table partitioned by date you need to detach partition first and then you're able to remove the data.
Lokk at DETACH section of https://clickhouse.yandex/docs/en/query_language/misc/

**Comment by q0987** [comment]
Hello,

I have tried the proposed method but it doesn't work for us.

```
:) alter table default.MyTable detach partition '2018-11-12'
ALTER TABLE default.MyTable
    DETACH PARTITION '2018-11-12'
Ok.
0 rows in set. Elapsed: 0.002 sec.
```

Based on my experience, the detached partitions should be moved into a directory as follows:
/var/lib/clickhouse/data/default/MyTable/detached

However, the detach operation does't do any work. Here is the file structure after I detach the failed to drop partition.

$ ls /var/lib/clickhouse/sx/data/default/MyTable/
drwxr-x--- 2 clickhouse clickhouse 36 Nov 12 17:11 20181112_6310_6310_0
drwxr-x--- 2 clickhouse clickhouse 36 Nov 12 17:37 20181112_6288_6309_1
drwxr-x--- 2 clickhouse clickhouse 36 Nov 12 17:46 20181112_6315_6319_1
drwxr-x--- 2 clickhouse clickhouse 36 Nov 12 17:46 20181112_6311_6314_1
drwxr-x--- 2 clickhouse clickhouse 36 Nov 12 17:59 20181112_6248_6251_1
drwxr-x--- 2 clickhouse clickhouse 36 Nov 12 18:02 20181112_6252_6287_1
drwxr-x--- 2 clickhouse clickhouse 36 Nov 12 18:16 20181112_6153_6247_3
drwxr-x--- 2 clickhouse clickhouse 36 Nov 13 17:33 20181112_6320_6334_2

$ ls /var/lib/clickhouse/sx/data/default/MyTable/detached

Also, I tried to attach this partition without success.
```
:) alter table default.MyTable attach partition '2018-11-12'
ALTER TABLE default.MyTable
    ATTACH PARTITION '2018-11-12'
Ok.
0 rows in set. Elapsed: 0.008 sec.
```

`:) select partition,name,database,table from system.parts where table='MyTable' and database='default' and name='20181112_6310_6310_0'`

```
SELECT
    partition,
    name,
    database,
    table
FROM system.parts
WHERE (table = 'MyTable') AND (database = 'default') AND (name = '20181112_6310_6310_0')
Ok.
0 rows in set. Elapsed: 0.003 sec.

```
In summary, this issue was caused by the failed partition drop failure.

2018.11.15 14:39:35.586802 [ 5 ] <Warning> default.MyTable (Data): Removing temporary directory /var/lib/clickhouse/data/default/MyTable/tmp_delete_20181112_6148_6152_1
2018.11.15 14:39:35.606459 [ 25 ] <Error> executeQuery: Poco::Exception. Code: 1000, e.code() = 2, e.displayText() = File not found: /var/lib/clickhouse/data/default/Mytable/tmp_delete_20181112_6148_6152_1/minmax_date.idx, e.what() = File not found (from 127.0.0.1:54484) (in query: alter table default.MyTable drop partition '2018-11-12')

ClickHouse client version 18.6.0.
Connecting to 127.0.0.1:9000 as user simplex_admin.
Connected to ClickHouse server version 18.6.0 revision 54401.

**Comment by alexey-milovidov** [comment]
> Question 1> Can I manually delete those directories 20181112_* since the system doesn't have knowledges about them and they are not in the detached directory.

You can manually delete the directory if DETACH was failed with `File not found` error and the partition does not exists in `system.parts`. It is safe because the server has already forgot about the data part.

But the issue `File not found` should not happen. It indicates concurrent attempts to remove the same directory. Recently I have fixed similar issue that can happen at the server startup. And your case helps me to realize that there is another issue to fix.

> Question 2> Is there a way that I can force the ClickHouseDB to scan the directory without restarting the server?

You can use `DETACH TABLE`  and `ATTACH TABLE` for this purpose.
